src/Engine/ProtoCore/Lang/CallSite.cs

```diff
@@ -2041,21 +1865,45 @@ public static StackValue PerformReturnTypeCoerce(FunctionEndPoint functionEndPoi
         #endregion
 
         /// <summary>
-        /// Returns a flat collection of ISerializable objects from a serialized representation of a SingleRunTraceData object.
+        /// Returns a flat collection of strings from a serialized representation of a SingleRunTraceData object.
         /// </summary>
         /// <param name="callSiteData">The serialized representation of a SingleRunTraceData object.</param>
-        /// <returns>A flat collection of ISerializable objects.</returns>
-        public static IList<ISerializable> GetAllSerializablesFromSingleRunTraceData(
+        /// <returns>A flat collection of strings.</returns>
+        public static IList<string> GetAllSerializablesFromSingleRunTraceData(
             RawTraceData callSiteData)
         {
-            var helper = TraceSerialiserHelper.FromCallSiteData(callSiteData.Data);
-            if (helper == null)
+            List<SingleRunTraceData> traceData = null;
+            try
+            {
+                //Optional Compression / Decompression
+                var data = DecompressSerializedTraceData(callSiteData.Data);
+                traceData = JsonConvert.DeserializeObject<List<SingleRunTraceData>>(data);
```

Comment:
perhaps it doesn't matter but I wonder if it might be more secure to use System.Text.Json - I think it has some different defaults when it comes to deserializing different types, I guess that could cause some issues for deserializing this class or other integrator specific trace classes...

https://github.com/DynamoDS/DynamoRevit/blob/master/src/Libraries/RevitServices/RevitTraceUtils.cs#L33

Reply:
I think it is easy to swap out to System.Text.Json.  I tried it but required fully qualified namespace but works exactly the same.

Reply:
Also we can update the RevitTraceUtils you reference above....  We will have to check with Vlad as he has removed that class in the NET7 D4R branch -> https://github.com/DynamoDS/DynamoRevit/blob/upgrade_to_net7/src/Libraries/RevitServices/RevitServices.csproj#L95